File: api/prisma/seed.ts
import { PrismaClient } from '@prisma/client'
import {
  RESTAURANTS,
  OPEN_SESSION,
  CLOSED_SESSION,
  type Session,
} from '@morningcart/shared'

const prisma = new PrismaClient()

// MenuItem ids collide across restaurants (short ids like 'foul','tea' repeat),
// so namespace them per restaurant to keep the DB primary key unique.
const itemDbId = (restaurantId: string, shortItemId: string): string =>
  `${restaurantId}__${shortItemId}`

// Office-local service date (Africa/Cairo) as 'YYYY-MM-DD'.
const cairoDate = new Intl.DateTimeFormat('en-CA', {
  timeZone: 'Africa/Cairo',
  year: 'numeric',
  month: '2-digit',
  day: '2-digit',
})

const serviceDate = (d: Date = new Date()): string => cairoDate.format(d)

// Price snapshot for an el-sobhy short item id (unitPrice on each OrderLine).
const elSobhy = RESTAURANTS.find((r) => r.id === 'el-sobhy')
if (!elSobhy) throw new Error("seed: missing 'el-sobhy' fixture restaurant")

const elSobhyPrice = (shortItemId: string): number => {
  const item = elSobhy.menu.find((m) => m.id === shortItemId)
  if (!item) throw new Error(`seed: no el-sobhy menu item for '${shortItemId}'`)
  return item.price
}

async function seedSessionOrders(
  sessionId: string,
  session: Session,
): Promise<number> {
  let lineCount = 0
  for (const o of session.orders) {
    const created = await prisma.order.create({
      data: { sessionId, person: o.person, paid: o.paid },
    })
    for (const line of o.lines) {
      await prisma.orderLine.create({
        data: {
          orderId: created.id,
          menuItemId: itemDbId('el-sobhy', line.itemId),
          qty: line.qty,
          note: line.note ?? null,
          forName: line.forName ?? null,
          unitPrice: elSobhyPrice(line.itemId),
        },
      })
      lineCount++
    }
  }
  return lineCount
}

async function main(): Promise<void> {
  // Idempotent reset — delete in FK-safe order (children first).
  await prisma.orderLine.deleteMany()
  await prisma.order.deleteMany()
  await prisma.session.deleteMany()
  await prisma.menuItem.deleteMany()
  await prisma.restaurant.deleteMany()

  // Restaurants + their menus, with stable namespaced ids.
  let restaurantCount = 0
  let menuItemCount = 0
  for (const r of RESTAURANTS) {
    await prisma.restaurant.create({
      data: {
        id: r.id,
        name: r.name,
        arabic: r.arabic ?? null,
        deliveryFee: r.deliveryFee,
      },
    })
    restaurantCount++

    for (const [index, item] of r.menu.entries()) {
      await prisma.menuItem.create({
        data: {
          id: itemDbId(r.id, item.id),
          restaurantId: r.id,
          name: item.name,
          arabic: item.arabic ?? null,
          price: item.price,
          kind: item.kind,
          sortOrder: index,
          available: true,
        },
      })
      menuItemCount++
    }
  }

  const today = serviceDate()

  // Open session — this morning, mid-flow (El Sobhy).
  const openSession = await prisma.session.create({
    data: {
      restaurantId: 'el-sobhy',
      startedBy: OPEN_SESSION.startedBy,
      status: 'open',
      serviceDate: today,
    },
  })

  // Closed session — an earlier day, finalized (El Sobhy).
  const yesterday = serviceDate(new Date(Date.now() - 24 * 60 * 60 * 1000))
  const closedSession = await prisma.session.create({
    data: {
      restaurantId: 'el-sobhy',
      startedBy: CLOSED_SESSION.startedBy,
      status: 'closed',
      serviceDate: yesterday,
      closedAt: new Date(),
    },
  })

  let lineCount = 0
  lineCount += await seedSessionOrders(openSession.id, OPEN_SESSION)
  lineCount += await seedSessionOrders(closedSession.id, CLOSED_SESSION)

  const sessionCount = 2
  const orderCount = OPEN_SESSION.orders.length + CLOSED_SESSION.orders.length

  console.log(
    `Seeded ${restaurantCount} restaurants, ${menuItemCount} menu items, ` +
      `${sessionCount} sessions, ${orderCount} orders, ${lineCount} order lines.`,
  )
}

main()
  .catch((err) => {
    console.error('Seed failed:', err)
    process.exitCode = 1
  })
  .finally(async () => {
    await prisma.$disconnect()
  })

File: api/breakfast/management/commands/seed.py
from datetime import datetime, timedelta
from zoneinfo import ZoneInfo

from django.conf import settings
from django.core.management.base import BaseCommand
from django.db import transaction
from django.utils import timezone

from breakfast.data import CLOSED_ORDERS, OPEN_ORDERS, RESTAURANTS, item_db_id
from breakfast.models import MenuItem, Order, OrderLine, Restaurant, Session


def service_date(offset_days: int = 0) -> str:
    d = datetime.now(ZoneInfo(settings.OFFICE_TZ)) - timedelta(days=offset_days)
    return d.strftime('%Y-%m-%d')


class Command(BaseCommand):
    help = 'Seed the database with demo restaurants and sessions. Skips silently if data exists (use --force to wipe and reseed).'

    def add_arguments(self, parser):
        parser.add_argument('--force', action='store_true', help='Wipe everything and reseed even if data exists.')

    @transaction.atomic
    def handle(self, *args, **options):
        if Restaurant.objects.exists() and not options['force']:
            self.stdout.write('Data already present — skipping seed (run with --force to wipe and reseed).')
            return
        # idempotent reset, FK-safe order
        OrderLine.objects.all().delete()
        Order.objects.all().delete()
        Session.objects.all().delete()
        MenuItem.objects.all().delete()
        Restaurant.objects.all().delete()

        el_sobhy_price = {}
        for r in RESTAURANTS:
            Restaurant.objects.create(id=r['id'], name=r['name'], arabic=r['arabic'], delivery_fee=r['delivery_fee'])
            for i, (short, name, arabic, kind, price) in enumerate(r['menu']):
                MenuItem.objects.create(
                    id=item_db_id(r['id'], short), restaurant_id=r['id'],
                    name=name, arabic=arabic, kind=kind, price=price, sort_order=i,
                )
                if r['id'] == 'el-sobhy':
                    el_sobhy_price[short] = price

        el_sobhy_fee = next(r['delivery_fee'] for r in RESTAURANTS if r['id'] == 'el-sobhy')

        def make_session(orders, status, sd, started_by):
            s = Session.objects.create(
                restaurant_id='el-sobhy', started_by=started_by, status=status,
                service_date=sd, delivery_fee=el_sobhy_fee,
                closed_at=(timezone.now() if status == 'closed' else None),
            )
            for person, paid, lines in orders:
                o = Order.objects.create(session=s, person=person, paid=paid)
                OrderLine.objects.bulk_create([
                    OrderLine(
                        order=o, menu_item_id=item_db_id('el-sobhy', short), qty=qty,
                        note=note, for_name=forn, unit_price=el_sobhy_price[short],
                    )
                    for (short, qty, note, forn) in lines
                ])

        make_session(OPEN_ORDERS, 'open', service_date(0), 'Omar')
        make_session(CLOSED_ORDERS, 'closed', service_date(1), 'Dina')

        counts = (
            Restaurant.objects.count(), MenuItem.objects.count(),
            Session.objects.count(), Order.objects.count(), OrderLine.objects.count(),
        )
        self.stdout.write(self.style.SUCCESS(
            'Seeded %d restaurants, %d menu items, %d sessions, %d orders, %d order lines.' % counts
        ))
